### ISSUE TITLE
YOLOv5+SimOTA from YOLOX

### ISSUE THREAD

**Issue by jay985735639** [ultralytics/yolov5#8451]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

SimOTA not only reduces the training time, but also avoids the extra hyperparameter problem in SinkhornKnopp algorithm.   SimOTA improved the AP of the detector from 45.0% to 47.3%, which was 3.0% higher than SOTA Ultralytis-YoloV3. Why not try to put SimOTA policies into YoloV5 ？

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
@jay985735639 thanks for the info. Can you point us to any relevant links/code examples for the implementation. Also, you marked that you're willing to submit a PR. Let's do that for better discussion?

**Comment by jay985735639** [comment]
https://blog.csdn.net/qq_34496674/article/details/124889638, I saw that a blog realized this idea and made a good map

**Comment by jay985735639** [comment]
I didn't get good results

**Comment by AyushExel** [comment]
OKay thanks for sharing.
@glenn-jocher This [link](https://blog.csdn.net/qq_34496674/article/details/124889638) has listed all modifications needed in yolov5 to enable simOTA and claims " The map is increased by 6% compared to the original yolov5" among other improvements.

**Comment by lixiangyussss** [comment]
> 我没有得到好的结果
你好可否加个联系方式来讨论simota

**Comment by glenn-jocher** [comment]
@AyushExel yes I'm aware of this. I experimented with a YOLOX simOTA-like label assignment strategy in build_targets() but was not able to produce mAP improvements, with the most notable change being ~50% longer training time. 

The basic idea is that instead of the current label assignment strategy which pairs all labels to targets within width-height multiple threshold (i.e. up to 4x larger or 4x smaller anchors in main cell and neighboring cells), the change would compute losses first among all of these and then only use the 10 lowest loss pairs to compute loss. The loss used is box+class loss, with an additional hyp to balance their contributions. 

More experimenting might be useful there. I think the branch is still around, but I can't find it right now. Need to cleanup the open branches.

https://arxiv.org/pdf/2107.08430.pdf
<img width="1087" alt="Screen Shot 2022-07-04 at 3 52 57 PM" src="https://user-images.githubusercontent.com/26833433/177168971-cb96e1bb-0980-4ab1-97ee-8e8c02e89a88.png">
